syntax = "proto3";

message GameObject {
  string url = 1;
  string types = 2;
  string name = 3;
  string desc_snippet = 4;
  string recent_reviews = 5;
  string all_reviews = 6;
  string release_date = 7;
  string developer = 8;
  string publisher = 9;
  string popular_tags = 10;
  string game_details = 11;
  string languages = 12;
  int32 achievements = 13;
  string genre = 14;
  string game_description = 15;
  string mature_content = 16;
  string minimum_requirements = 17;
  string recommended_requirements = 18;
  string original_price = 19;
  string discount_price = 20;
  string token = 21;
}

message AddGameResponse {
  string message = 1;
}

message UpdateGameResponse {
  string message = 1;
}

message DeleteGameRequest {
  string url = 1;
  string token = 2;
}

message DeleteGameResponse {
  string message = 1;
}

message DeleteUserRequest {
  string id = 1;
  string token = 2;
}

message DeleteUserResponse {
  string message = 1;
}

service AdminOperations {
  rpc AddGame (GameObject) returns (AddGameResponse);
  rpc UpdateGame (GameObject) returns (UpdateGameResponse);
  rpc DeleteGame (DeleteGameRequest) returns (DeleteGameResponse);
  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
}



